data-loader/core/src/main/java/com/scalar/db/dataloader/core/DataLoaderError.java

```diff
@@ -0,0 +1,292 @@
+  LOG_DIRECTORY_CREATION_FAILED(
+      Category.USER_ERROR, "0041", "Failed to create the log directory: %s", "", ""),
+  INVALID_CONTROL_FILE(Category.USER_ERROR, "0042", "Failed to parse the control file: %s", "", ""),
+  DIRECTORY_WRITE_ACCESS(
+      Category.USER_ERROR,
+      "0043",
+      "No permission to create or write files in the directory: %s",
+      "",
+      ""),
+  DIRECTORY_CREATION_FAILED(
+      Category.USER_ERROR, "0044", "Failed to create the directory: %s", "", ""),
+  PATH_IS_NOT_A_DIRECTORY(
+      Category.USER_ERROR, "0045", "Path exists but is not a directory: %s", "", ""),
+  FILE_PATH_IS_BLANK(Category.USER_ERROR, "0046", "File path must not be blank", "", ""),
+  FILE_NOT_FOUND(Category.USER_ERROR, "0047", "File not found: %s", "", ""),
+  INVALID_DATE_TIME_FOR_COLUMN_VALUE(
+      Category.USER_ERROR,
+      "0048",
+      "Invalid date time value '%s' specified for column %s in table %s in namespace %s",
+      "",
+      ""),
+  NULL_OR_EMPTY_KEY_VALUE_INPUT(
+      Category.USER_ERROR, "0049", "Key-value cannot be null or empty", "", ""),
+  INVALID_KEY_VALUE_INPUT(Category.USER_ERROR, "0050", "Invalid key-value format: %s", "", ""),
+  SPLIT_INPUT_VALUE_NULL(Category.USER_ERROR, "0051", "Value must not be null", "", ""),
+  SPLIT_INPUT_DELIMITER_NULL(Category.USER_ERROR, "0052", "Delimiter must not be null", "", ""),
+  CONFIG_FILE_PATH_BLANK(Category.USER_ERROR, "0053", "Config file path must not be blank", "", ""),
+  INVALID_DATA_CHUNK_SIZE(
+      Category.USER_ERROR, "0054", "Data chunk size must be greater than 0", "", ""),
+  INVALID_TRANSACTION_SIZE(
+      Category.USER_ERROR, "0055", "Transaction size must be greater than 0", "", ""),
+  INVALID_MAX_THREADS(
+      Category.USER_ERROR, "0056", "Number of max threads must be greater than 0", "", ""),
+
+  //
+  // Errors for the internal error category
+  //
+  ERROR_CRUD_EXCEPTION(
+      Category.INTERNAL_ERROR,
+      "0000",
+      "Something went wrong while trying to save the data. Details: %s",
+      "",
+      ""),
+  ERROR_SCAN(
+      Category.INTERNAL_ERROR,
+      "0001",
+      "Something went wrong while scanning. Are you sure you are running in the correct transaction mode? Details: %s",
+      "",
+      ""),
+  CSV_FILE_READ_FAILED(
+      Category.INTERNAL_ERROR, "0002", "Failed to read CSV file. Details: %s", "", ""),
+  CSV_FILE_HEADER_READ_FAILED(
+      Category.INTERNAL_ERROR, "0003", "Failed to CSV read header line. Details: %s", "", ""),
+  DATA_CHUNK_PROCESS_FAILED(
+      Category.INTERNAL_ERROR,
+      "0004",
+      "Data chunk processing was interrupted. Details: %s",
+      "",
+      ""),
+  JSON_FILE_READ_FAILED(
+      Category.INTERNAL_ERROR, "0005", "Failed to read JSON file. Details: %s", "", ""),
+  JSONLINES_FILE_READ_FAILED(
+      Category.INTERNAL_ERROR, "0006", "Failed to read JSON Lines file. Details: %s", "", ""),
+  ;
+
+  private static final String COMPONENT_NAME = "DB-DATA-LOADER";
```

Comment:
@feeblefakie The prefix of the error code for data-loader is `DB-DATA-LOADER`. Is that okay?